### ISSUE TITLE
GameGuru Max - Request - Enhancement - Terrain Size

### ISSUE THREAD

**Issue by Lemuripest93** [TheGameCreators/GameGuruRepo#3433]
I imagine a large portion of people are probably okay with the current terrain size, however i have a few concepts that would require the terrain size to be a lot bigger. i'll be using a prototype i begun working on last night as an example, but before that, i also have an open-world concept in my headlights that would add least require the 5.0km terrain limit to double to 10km, though preferably the terrain limit will even surpass that.

In the case of performance issues (because i realize there might be some concern in regards to that,) the engine could warn you (like a warning sign above the slider) that you're surpassing the capabilities of your system specs by increasing the terrain past a certain limit. 

What i'm working on is a miniaturization (toy soldier) concept for a single-player/multi-player, the SP portion being an open-world survival type game whilst the MP portion plays a lot like Battlefield 2 and on-wards, what you see in the image below is what i was currently capable of filling an empty 5km map with for this concept, ideally i'd be able to make a small street with maybe 8 houses max for the single-player portion.

Now i would not be able to fully work on this anyway until Max is probably in version 2 maybe version 1 if i'm lucky? the actual multi-player aspect has not been added yet, Not to mention of course a lot of performance updates which makes your computer less likely to spontaneously combust.

![ggmterrlim](https://user-images.githubusercontent.com/104637906/178342149-e85fb695-e865-436e-9bf8-4e57c29ecefc.png)

(I'm currently uploading a video to showcase my crappy draft of the map above just to give a more overall idea of the scale, this comment will be updated once YouTube is done processing the video in HD.)

**Comment by Lemuripest93** [comment]
Here is a video demonstration showcasing large portion of the map above for scale reference: https://www.youtube.com/watch?v=xS9pNIkCiw0

Not necessarily relevant to others, but i thought it would be fun to showcase anyway, feel free to skip around in the video since there's a lot of running.

**Comment by Necrym59** [comment]
The actual navmesh play area and terrain size are locked if you want a bigger area then it will have to load somehow, ALL games do this one way or another either procedurally or progressively in either case none load the entire 110 square mile maps in in one hit. Load a map when you hit a marker and make the opening of the second map a continuation is the closest bet for now hopefully map edge loading will be implemented later.

**Comment by earthling45** [comment]
@Lemuripest93 something similar i thought about last week, let's make a game in which the player is an ant and must avoid being stepped on by the AI which is walking/playing around in the garden of course. :)

**Comment by Lemuripest93** [comment]
@Necrym59 Yeah, hopefully it will, not really worried about it right now - just thought it would be good to post this early so the process of working towards a solution can also happen a lot earlier.

@earthling45 There is an interesting game from back in the day on the Wii called 'Deadly Creatures' that i think did this, but you played as a spider? anyways this concept was inspired by the more recent game 'Hypercharge Unboxed' and the classic 90's movie 'Small Soldiers'.

**Comment by fearlesswee** [comment]
I agree that the procedural terrain system is being a bit under-used atm, would be awesome to allow for larger levels, but performance is a BIG concern. MAX still loads the entire level into memory at full detail all at once, which if you had a 10km (or heaven forbid larger) map full of assets, it would be a MASSIVE performance/memory hog. Lee mentioned in passing a few times about looking into adding a system that streams texture data from disk as the player moves around the world to keep memory usage down and performance high, but imo that hypothetical system would likely have to extend into also streaming entity geometry and such for a 10km or even larger map size to work.

(Btw, very interesting game idea! A toy soldier open world survival game sounds pretty unique, personally reminds me a bit of the survival game "Grounded" where it's more like "honey I shrunk the kids" but as a survival crafting game.)

**Comment by Lemuripest93** [comment]
@fearlesswee For my particular concept to work there'd need to be a lot of things added, i'm toying around (pun totally intended) with the idea of dismemberment and basically looting dead toy soldiers for body parts and switching things out for higher durability parts, or maybe parts with different functions.. say an arm with a chaingun attachment? but it shoots rubber pellets, or spring loaded legs that allow for super-jumps, things like that.

You could also reattach your own blown off and broken body parts with superglue or other items but with 50% or less durability. basically you are carrying your blown off bits in an inventory, and then find different supplies in the game world to repair yourself with.

MP section would have vehicular combat like dogfighing in RC planes, blowing each other up in toy tanks, riding beetles into battle and using toothpicks for jousting, all sort of silly but fun stuff like that, game modes would include classics like capture the flag, team death match, a sort of battle royale mode with a few twists, a classic battlefield type skirmishes where you take over and hold areas until the map is completely yours, and a sandbox mode where you'd be able to spawn in different items, bots, vehicles and just mess around with your friends.

Whether or not this is actually possible a lot further down the line in Max at all is a different question, but we'll see.

**Comment by earthling45** [comment]
@Lemuripest93 haven't seen that movie, though i wonder why you don't go the other way by shrinking the player character? isn't that possible?

**Comment by davetgc** [comment]
Most systems won't cope with landscapes of such a size, clever game design is often needed to give the impression of a massive playing area,
